### DISCUSSION TITLE
Custom field attributes (for easier testing & styling)

### DISCUSSION THREAD

**Discussion by emahuni** [directus/directus#18907] [Feature Requests]
### Summary

To increase its customizability I suggest that there be a way to add custom field attributes to make it easier to test and customize fields using css selectors. These attributes will be added to elements in the DOM.

### Basic Example

The following config 

![image](https://github.com/directus/directus/assets/7290508/4ebac263-2f6a-409a-98d4-a1586c3846b4)

adds these to the DOM element when rendered:

![image](https://github.com/directus/directus/assets/7290508/10598c52-d7c5-4e49-9702-8d5382ae92aa)

### Motivation

I use end to end testing with Cypress and it's not easy to try and select the appropriate field for automated testing. Also, there is a need to style certain fields differently and selecting them isn't easy as well.

### Detailed Design

This can be done in the Vue files that create the interfaces and displays. Binding of the custom attributes using v-bind should be a no-brainer.

- a json interface can be used to enter this info
- each entry in the array is: key=attribute and prop=attribute value
- entries with empty values are booleans like `disabled` as in `<button disabled></button>

### Requirements List

__

### Drawbacks

__

### Alternatives

___

### Adoption Strategy

__

### Unresolved Questions

- should it be the same for both interface and display or each should have their own custom attributes entries?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
